lib/Graph/Graph.cpp

```diff
@@ -820,20 +820,29 @@ ConcatNode *Function::createConcat(llvm::StringRef name,
   return addNode(new ConcatNode(name, OT, ops, dimension));
 }
 
-InsertTensorNode *Function::createTile(llvm::StringRef name, NodeValue input,
-                                       unsigned_t tiles, unsigned_t axis) {
+TileNode *Function::createTile(llvm::StringRef name, NodeValue input,
+                               unsigned_t tiles, unsigned_t axis) {
   assert(tiles > 0 && "Tiles must be non-zero.");
   assert(axis >= 0 && axis < input.dims().size() &&
          "Axis must fall in range of source dims.");
 
   ShapeVector outShape(input.dims().begin(), input.dims().end());
   outShape[axis] *= tiles;
   auto OT = getParent()->uniqueTypeWithNewShape(input.getType(), outShape);
+  return addNode(new TileNode(name, OT, input, tiles, axis));
+}
+
+InsertTensorNode *Function::createInsertTensor(llvm::StringRef name,
+                                               NodeValue big, NodeValue small,
+                                               llvm::ArrayRef<size_t> start,
+                                               unsigned_t count,
+                                               unsigned_t axis) {
+  assert(big.dims().size() == small.dims().size() &&
```

Comment:
I realized that we check this in `InsertTensorNode::verify()`. But we don't assert anything related to `count` and `axis` there.

Maybe makes sense to remove checks here and improve checks there?

Reply:
Yeah it seems this partially duplicates the checks from `verify()`. I removed them from here and added `axis`/`count` checks to InsertNode.